src/js/inject.js

```diff
@@ -465,6 +511,151 @@ $('body').on('click', '#open-modal', (ev) => {
+  const iframeSrc = $(embed).find('iframe').attr('src');
+  if (iframeSrc && iframeSrc.length) {
+    return getQueryVariable(iframeSrc, 'video_url');
+  }
+  return '';
+};
+
+const getMedia = (elem) => {
+  const media = [];
+
+  // check the obscure case of being in a modal
+  const modalContainer = $(elem).parents('.js-modal-panel');
+
+  $(modalContainer).find('.js-embeditem').each((i, embed) => {
+    const iframeSrc = getMediaFromEmbed(embed);
+    if (iframeSrc.length) {
+      media.push(iframeSrc);
+    }
+  });
+  if (media.length) return media;
+
+  // we want to be sure this modal is a gif
+  $(modalContainer).parents('[data-btd-provider="gif"]')
+    .find('.btd-embed-container video').each((i, link) => {
+      const vidSrc = $(link).attr('src');
+      if (vidSrc.length) {
+        media.push(vidSrc);
+      }
+    });
+  if (media.length) return media;
+
+  // check the most typical case of a stream item
+  const normalContainer = $(elem).parents('article');
+
+  $(normalContainer).find('.quoted-tweet').each((i, link) => {
+    const tweetID = $(link).data('tweet-id');
+    const fullURI = `${$(link).find('.account-link').attr('href')}/status/${tweetID}`;
+    if (fullURI.length) {
+      media.push(fullURI);
+    }
+  });
+  if (media.length) return media;
+
+  $(normalContainer).find('.tweet-detail-media .js-media iframe').parent().each((i, embed) => {
+    const iframeSrc = getMediaFromEmbed(embed);
+    if (iframeSrc.length) {
+      media.push(iframeSrc);
+    }
+  });
+  if (media.length) return media;
+
+  $(normalContainer).find('.is-video .js-media-image-link').each((i, link) => {
+    const oTarget = $(link).attr('target');
+    const oSrc = $(link).attr('src');
+    $(link).attr('target', '');
+    $(link).attr('src', '#');
+    $(link).click();
+
+    const embeds = $('.js-embeditem');
+    $(embeds).each((j, embed) => {
+      const iframeSrc = getMediaFromEmbed(embed);
+      if (iframeSrc.length) {
+        media.push(iframeSrc);
+      }
+      $('.mdl-dismiss .icon-close').click();
+    });
+
+    $(link).attr('target', oTarget);
+    $(link).attr('src', oSrc);
+  });
+  if (media.length) return media;
+
+  $(normalContainer).find('video.js-media-gif').each((i, link) => {
+    const vidSrc = $(link).attr('src');
+    if (vidSrc.length) {
+      media.push(vidSrc);
+    }
+  });
+  if (media.length) return media;
+
+  $(normalContainer).find('.js-media ')
+    .find('.med-link, .media-image')
+    .each((i, link) => {
+      const imgSrc = cleanMediaUrl($(link).css('background-image'));
+
+      if (imgSrc && imgSrc.length) {
+        media.push(imgSrc);
+      }
+    });
+
+  // these rules work in all contexts
+  const all = $()
+    .add($(normalContainer))
+    .add($(modalContainer));
+
+  $(all).find('img.media-img').each((i, link) => {
+    const imgSrc = cleanMediaUrl($(link).attr('src'));
+
+    if (imgSrc.length) {
+      media.push(imgSrc);
+    }
+  });
+
+  return media;
+};
```

Comment:
I feel like the whole function is a bit more complicated than it should be. Since it looks like we're restricting ourselves to Twitter media stuff, we can easily do something way simpler by doing the following:

1. start from the target
2. get the closest parent node with the `data-key` attribute
3. get the value of said attribute
4. query the DOM with a query like `[data-column] [data-key=value]`
5. get the value of `data-column`
6. call the [`getChirpFromKey`](http://github.com/eramdam/BetterTweetDeck/blob/1b1bf86bd4c007865bcb4a3fbe544035b608a7cd/src/js/inject.js#L11-L60) method
7. inspect the actual tweet model to get our media URLs
8. ??? 
9. PROFIT

This should reduce by _a lot_ the DOM access and should be easier to maintain given that we're almost independant from the DOM layout.

Feel free to ask me for help if you're struggling on something, use the [debug functions](http://github.com/eramdam/BetterTweetDeck/blob/1b1bf86bd4c007865bcb4a3fbe544035b608a7cd/src/js/inject.js#L62-L80) while developing, they should come handy.